pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go

```diff
@@ -0,0 +1,288 @@
+		}))
+		if err != nil {
+			return err
+		}
+	}
+
+	recentBackupCondition, err := c.ensureRecentBackup(ctx, status, syncCtx.Recorder())
+	if err != nil {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "UpgradeBackupControllerDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "Error",
+			Message: err.Error(),
+		}))
+		if updateErr != nil {
+			syncCtx.Recorder().Warning("ClusterBackupControllerUpdatingStatus", updateErr.Error())
+		}
+		return err
+	}
+	_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient,
+		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:   "UpgradeBackupControllerDegraded",
+			Status: operatorv1.ConditionFalse,
+			Reason: "AsExpected",
+		}))
+	if updateErr != nil {
+		syncCtx.Recorder().Warning("ClusterBackupControllerUpdatingStatus", updateErr.Error())
+		return updateErr
+	}
+
+	// If recentBackupCondition is nil work is done and no need to update status.
+	if recentBackupCondition == nil {
+		return nil
+	}
+
+	_, _, err = v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(*recentBackupCondition))
+	if err != nil {
+		syncCtx.Recorder().Warning("ClusterBackupControllerUpdatingStatus", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+// ensureRecentBackup ensures that a new backup pod is created if one does not exist, returns the appropriate RecentBackup condition.
+func (c *UpgradeBackupController) ensureRecentBackup(ctx context.Context, operatorStatus *operatorv1.OperatorStatus, recorder events.Recorder) (*operatorv1.OperatorCondition, error) {
+	clusterVersion, err := c.clusterVersionLister.Get("version")
+	if err != nil {
+		return nil, err
+	}
+
+	// Check cluster version status for backup condition.
+	if !isRequireRecentBackup(clusterVersion) {
+		return nil, nil
+	}
+
+	recentBackupName := fmt.Sprintf("upgrade-backup-%s", time.Now().Format("2006-01-02_150405"))
+
+	backupPod, err := c.podLister.Pods(operatorclient.TargetNamespace).Get(clusterBackupPodName)
+	// No backup found, attempt to create one.
+	if apierrors.IsNotFound(err) {
+		// Check nodes for backup preconditions
+		backupNodeName, err := c.getBackupNodeName()
+		if err != nil {
+			return nil, err
+		}
+
+		if err := createBackupPod(ctx, backupNodeName, recentBackupName, c.operatorImagePullSpec, c.targetImagePullSpec, c.kubeClient.CoreV1(), recorder); err != nil {
+			return nil, fmt.Errorf("pod/%s: %v", clusterBackupPodName, err)
+		}
+		recorder.Eventf("UpgradeBackupInProgress", "Backup pod created on node: %s", backupNodeName)
+
+		return &operatorv1.OperatorCondition{
+			Status:  operatorv1.ConditionUnknown,
+			Type:    backupConditionType,
+			Reason:  "UpgradeBackupInProgress",
+			Message: fmt.Sprintf("Upgrade backup pod created on node %s", backupNodeName),
+		}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// Check the existing Pod Status and update condition respectively.
```

Comment:
This logic does a bit more, and this msg is a bit misleading.

```suggestion
	// Check the existing Pod Status, reconcile pod and update condition respectively.
```